extensions/notification/utils/config.js

```diff
@@ -0,0 +1,89 @@
+const constants = require('../constants');
+
+/**
+ * Transforms array of configurations to a Map
+ * @param  {Object[]} configs - Array of bucket notification configurations
+ * @return {Map} Map of bucket notification configurations
+ */
+function transform(configs) {
+    const configMap = new Map();
+    configs.forEach(config => {
+        if (config.bucket && config.notificationConfiguration) {
+            const bucket = config.bucket;
+            const conf = config.notificationConfiguration.queueConfigurations;
+            if (configMap.has(bucket)) {
+                const val = configMap.get(bucket);
+                configMap.set(bucket, [...val, ...conf]);
+            } else {
+                configMap.set(bucket, [...conf]);
+            }
+        }
+    });
+    return configMap;
+}
+
+/**
+ * Validate object name against bucket notification configuration filters
+ * @param  {Object[]} filterRules - Filter rules from the bucket notification
+ * configuration.
+ * @param  {string} objectName - Name of the object
+ * @return {boolean} true if filter matches the object name
+ */
+function validateFilter(filterRules, objectName) {
+    return filterRules.some(rule => {
+        const name = rule.name;
+        const value = rule.value;
+        if (name === constants.nameFilter.prefix) {
+            return objectName.startsWith(value);
+        }
+        if (name === constants.nameFilter.suffix) {
```

Comment:
Is it allowed to have both `prefix` and `suffix` in the same rule? If so, you would need to change the logic to test each in turn. To accept as many criteria as needed you can filter out by negating the test, so e..g `if (!objectName.startsWith(value)) { return false; }`

Reply:
Yes, both conditions are allowed in a configuration.